lib/item.py

```diff
@@ -391,17 +405,20 @@ def _castvalue_to_itemtype(self, value, compat):
         """
         # casting of value, if compat = latest
         if compat == ATTRIB_COMPAT_LATEST:
-            mycast = globals()['_cast_' + self._type]
-            try:
-                value = mycast(value)
-            except:
-                logger.warning("Item {}: Unable to cast '{}' to {}".format(self._path, str(value), self._type, str(mycast)))
-                if isinstance(value, list):
-                    value = []
-                elif isinstance(value, dict):
-                    value = {}
-                else:
-                    value = mycast('')
+            if self._type != None:
+                mycast = globals()['_cast_' + self._type]
+                try:
+                    value = mycast(value)
+                except:
+                    logger.warning("Item {}: Unable to cast '{}' to {}".format(self._path, str(value), self._type, str(mycast)))
+                    if isinstance(value, list):
+                        value = []
+                    elif isinstance(value, dict):
+                        value = {}
+                    else:
+                        value = mycast('')
+            else:
+                logger.warning("Item {}: Unable to cast '{}' to {}".format(self._path, str(value), self._type))
```

Comment:
This message is logged when `self._type` is `None`. Is this message helpful in this case? Maybe it should clarify that the type is wrong and that's why it can't be casted.